homeassistant/components/squeezebox/media_player.py

```diff
@@ -128,16 +129,19 @@
         ipaddr = socket.gethostbyname(host)
     except (OSError) as error:
         _LOGGER.error("Could not communicate with %s:%d: %s", host, port, error)
-        return False
+        raise PlatformNotReady
 
     if ipaddr in known_servers:
         return
 
-    known_servers.add(ipaddr)
     _LOGGER.debug("Creating LMS object for %s", ipaddr)
     lms = LogitechMediaServer(hass, host, port, username, password)
 
     players = await lms.create_players()
+    if players is None:
+        raise PlatformNotReady
```

Comment:
Do you think that happens if we can connect to the box without connection error?

Reply:
No. `None` returned only in [one case](https://github.com/home-assistant/home-assistant/blob/a711b0925902924cfee9ec0a6784644b05c5939f/homeassistant/components/squeezebox/media_player.py#L201)